data_structure/segment_trees.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2020 @author Lasha Bukhnikashvili
+ * @file
+ * Program describes usage of a popular Data Structure: Segment Trees
+ * in Finding maximum number and its index on a given Interval
+ * in logarithmic = Log(N) Time using SegmentTree Data Structures
+ * with PreProcessing O(N * Log(N)) Time
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+/**
+ * Each Node contains corresponding value
+ * in array and it's index
+ */
+struct Node {
+    int val;
+    int ind;
+} null;
+
+/**
+ * Data Structure stores initial array and its nodes,
+ * which has the 4th size of the array = 4*N
+ */
+struct SegmentTree {
+    int N;
+    std::vector<int> arr;
+    std::vector<Node> nodes;
+} ST;
+
+/**
+ * This is a recursive method, which
+ * Builds the Tree in ST struct variable.
+ * It should be called only once for each array.
+ * Works in O(N * log N)
+ */
```

Comment:
please add description of input function arguments. 
what do `v`, `tl`, `tr` represent? please check the [reference](https://www.doxygen.nl/manual/docblocks.html) and example [code](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/math/complex_numbers.cpp) from the repo that does not use global variables, all functions and variables are documented and organized into class structure and modular in implementation.